plugins/flatpak/gs-self-test.c

```diff
@@ -1398,6 +1399,11 @@ gs_plugins_flatpak_app_update_func (GsPluginLoader *plugin_loader)
 				  G_CALLBACK (update_app_progress_notify_cb),
 				  &progress_cnt);
 
+	/* check that the runtime is not the update's one */
```

Comment:
I'd personally add a new test (e.g. `gs_plugins_flatpak_app_update_runtime_changed_func` or similar) instead of adapting the currently existing test case, since you're testing different use cases, and both should be checked.

Reply:
The tests are actually not trivial to implement currently, so I preferred to update the current test, as also the difference won't make any of previous tests in this function invalid.
It's a complex update and I think it's fine to test it in the current update test.

Reply:
The difference is that you're testing different things: before you were testing updates that didn't require a different runtime, now you're testing updates that require a different runtime. I'm all for adding new tests everywhere, but not at the expense of stopping to test things that were tested before.